pkg/nodeipam/ipam/range_allocator.go

```diff
@@ -309,34 +308,34 @@ func (r *rangeAllocator) ReleaseCIDR(node *corev1.Node) error {
 
 // Marks all CIDRs with subNetMaskSize that belongs to serviceCIDR as used across all cidrs
 // so that they won't be assignable.
-func (r *rangeAllocator) filterOutServiceRange(serviceCIDR *net.IPNet) {
+func filterOutServiceRange(clusterCIDRs []*net.IPNet, cidrSets []*cidrset.CidrSet, serviceCIDR *net.IPNet) {
 	// Checks if service CIDR has a nonempty intersection with cluster
 	// CIDR. It is the case if either clusterCIDR contains serviceCIDR with
 	// clusterCIDR's Mask applied (this means that clusterCIDR contains
 	// serviceCIDR) or vice versa (which means that serviceCIDR contains
 	// clusterCIDR).
-	for idx, cidr := range r.clusterCIDRs {
+	for idx, cidr := range clusterCIDRs {
 		// if they don't overlap then ignore the filtering
 		if !cidr.Contains(serviceCIDR.IP.Mask(cidr.Mask)) && !serviceCIDR.Contains(cidr.IP.Mask(serviceCIDR.Mask)) {
 			continue
 		}
 
 		// at this point, len(cidrSet) == len(clusterCidr)
-		if err := r.cidrSets[idx].Occupy(serviceCIDR); err != nil {
-			klog.Errorf("Error filtering out service cidr out cluster cidr:%v (index:%v) %v: %w", cidr, idx, serviceCIDR, err)
+		if err := cidrSets[idx].Occupy(serviceCIDR); err != nil {
+			klog.Errorf("Error filtering out service cidr out cluster cidr:%v (index:%v) %v: %v", cidr, idx, serviceCIDR, err)
 		}
 	}
 }
 
 // updateCIDRsAllocation assigns CIDR to Node and sends an update to the API server.
 func (r *rangeAllocator) updateCIDRsAllocation(data nodeReservedCIDRs) error {
 	var err error
-	var node *corev1.Node
+	var node *v1.Node
 	defer r.removeNodeFromProcessing(data.nodeName)
 	cidrsString := cidrsAsString(data.allocatedCIDRs)
 	node, err = r.nodeLister.Get(data.nodeName)
 	if err != nil {
-		klog.Errorf("Failed while getting node %v for updating Node.Spec.PodCIDRs: %w", data.nodeName, err)
+		klog.Errorf("Failed while getting node %v for updating Node.Spec.PodCIDRs: %v", data.nodeName, err)
```

Comment:
do we need to skip non-existing node here?

Reply:
how do we tell if a node is existed or not? by checking if the providerID is good?